onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -123,10 +124,27 @@ using namespace xnnpack;
 
 XnnpackExecutionProvider::XnnpackExecutionProvider(const XnnpackExecutionProviderInfo& info)
     : IExecutionProvider{kXnnpackExecutionProvider, true} {
-  if (info.xnn_thread_pool_size > 1) {
-    // pthreadpool is independent of ort-threadpoool, so we have to disable cpu spinning for ort-threadpool.
-    // otherwise, the pthreadpool will be starved and harm performance a lot.
-    xnnpack_thread_pool_ = pthreadpool_create(static_cast<size_t>(info.xnn_thread_pool_size));
+  int xnn_thread_pool_size = info.xnn_thread_pool_size;
+  int ort_thread_pool_size = info.session_options ? info.session_options->intra_op_param.thread_pool_size : 1;
+  bool allow_intra_op_spinning =
+      info.session_options && info.session_options->config_options.GetConfigOrDefault(
```

Comment:
Is it expected to turn off spinning if session options is null? In the regular inferencing workflow spinning is turned on by default.

Reply:
Thanks. It's reasonable to keep alignment with the default value.